server/lib/config/nginx.conf

```diff
@@ -0,0 +1,141 @@
+#+
+# Nginx configuration for the Pbench Server.
+#-
+
+user              nginx;
+worker_processes  auto;
+pid               /run/nginx.pid;
+
+# All nginx error logs flow through syslog.
+error_log         syslog:server=unix:/dev/log,facility=local5,tag=nginx,nohostname  info;
+
+# Load dynamic modules. See /usr/share/doc/nginx/README.dynamic.
+include           /usr/share/nginx/modules/*.conf;
+
+events {
+    worker_connections  1024;
+}
+
+http {
+    log_format  main  escape=json  '@cee:{"nginx":{"v0":{'
+        '"body_bytes_sent": $body_bytes_sent, '
+        '"bytes_sent": $bytes_sent, '
+        '"connection": $connection, '
+        '"connection_requests": $connection_requests, '
+        '"http": {'
+            '"referer": "$http_referer", '
+            '"user_agent": "$http_user_agent", '
+            '"x_forwarded_for": "$http_x_forwarded_for"'
+        '}, '
+        '"msec": $msec, '
+        '"pipe": "$pipe", '
+        '"remote": {'
+            '"addr": "$remote_addr", '
+            '"port": "$remote_port", '
+            '"user": "$remote_user"'
+        '}, '
+        '"request": {'
+            '"args": "$args", '
+            '"length": $request_length, '
+            '"method": "$request_method", '
+            '"raw": "$request", '
+            '"time": $request_time, '
+            '"uri": "$uri"'
+        '}, '
+        '"server": {'
+            '"name": "$server_name", '
+            '"port": "$server_port"'
+        '}, '
+        '"status": $status, '
+        '"time": "$time_iso8601", '
+        '"upstream": {'
+            '"connect_time": "$upstream_connect_time", '
+            '"header_time": "$upstream_header_time", '
+            '"response_time": "$upstream_response_time"'
+        '}'
+    '}}}';
+
+    # All nginx access logs flow through syslog as well.
+    access_log  syslog:server=unix:/dev/log,facility=local6,tag=nginx,severity=info,nohostname  main;
+
+    sendfile             on;
+    tcp_nopush           on;
+    tcp_nodelay          on;
+    keepalive_timeout    65;
+    types_hash_max_size  4096;
+
+    include              /etc/nginx/mime.types;
+    default_type         application/octet-stream;
+
+    server {
+        listen       8080;
+        listen       [::]:8080;
+        server_name  _;
+        root         /srv/pbench/public_html;
+        index        index.html;
+
+        error_page   404              /404.html;
+        error_page   500 502 503 504  /50x.html;
+        error_page   597              @maintenance;
+
+        location @maintenance {
+            rewrite ^(.*)$ /maintenance.json break;
+        }
+
+        location = /404.html {
+            internal;
+        }
+
+        location = /50x.html {
+            internal;
+        }
+
+        location /api {
```

Comment:
Why do lines 85 and 89 include equal signs when the other lines (e.g., 81 and 93) do not?

Reply:
This is an Nginx [behavior](https://nginx.org/en/docs/http/ngx_http_core_module.html#location): `=` means "match that URI in its entirety", while no `=` means a prefix, essentially.